Cargo.nix

```diff
@@ -0,0 +1,3487 @@
+let
+  profiles = {
+  };
+in
+args@{
+  release ? true,
+  rootFeatures ? { 
+    "cargo2nix2/default" = { };
+  },
+}:
+{ rustPackages, buildRustPackages, mkRustCrate, buildPlatform, hostPlatform, rustLib, lib }:
+let
+  inherit (rustLib) fetchCratesIo fetchCrateLocal fetchCrateGit expandFeatures;
+  rootFeatures = expandFeatures args.rootFeatures;
+  outerHostPlatform = hostPlatform;
+in
+{
+  "adler32 1.0.4 registry+https://github.com/rust-lang/crates.io-index" = args@{ hostPlatform ? outerHostPlatform, ... }: mkRustCrate (args // {
+    inherit release profiles;
+    packageId = "adler32 1.0.4 registry+https://github.com/rust-lang/crates.io-index";
+    name = "adler32";
+    version = "1.0.4";
+    registry = "registry+https://github.com/rust-lang/crates.io-index";
+    src = fetchCratesIo {
+      name = "adler32";
+      version = "1.0.4";
+      sha256 = "5d2e7343e7fc9de883d1b0341e0b13970f764c14101234857d2ddafa1cb1cac2";
+    };
+    features = builtins.concatLists [
+    ];
+    dependencies = {
+    };
+    devDependencies = {
+    };
+    buildDependencies = {
+    };
+    manifest = builtins.fromTOML "[package]\nauthors = [\"Remi Rampin <remirampin@gmail.com>\"]\ndescription = \"Minimal Adler32 implementation for Rust.\"\ndocumentation = \"https://remram44.github.io/adler32-rs/index.html\"\nkeywords = [\"adler32\", \"hash\", \"rolling\"]\nlicense = \"Zlib\"\nname = \"adler32\"\nreadme = \"README.md\"\nrepository = \"https://github.com/remram44/adler32-rs\"\nversion = \"1.0.4\"\n";
```

Comment:
I'm curious: do we even use these attributes for anything in any stage of `cargo2nix`? I know the original `Cargo.lock` doesn't even include these fields at all, so perhaps we can omit this key altogether?

Reply:
We do. `cargo` needs the manifest to provide [these environment variables](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates) during compilation.

Reply:
If the crate source itself is being vendored in the Nix store via `fetchCratesIo`, presumably including the `Cargo.toml`, wouldn't you be able to read the contents of the file from disk? Why does it need to be embedded directly into the `Cargo.nix`?

Reply:
The source we get from `fetchCratesIo` is a tarball, which won't be unpacked until the build runs. We need the manifest before the build to prepare a patched manifest. We can also do this in bash during the build, but it's a bit more complicated. I'll look into this.

Reply:
Got it! I see why it would be simpler to have it embedded in the `Cargo.nix` at generation time, but then again, it would be great to get rid of the extra "fluff" in the generated Nix code. It's definitely not a priority to remove it, but it sure would be cool to do so down the line. 😄